### ISSUE TITLE
Add git submodules for picongpu and possible other required projects

### ISSUE THREAD

**Issue by n01r** [ComputationalRadiationPhysics/parataxis#11]
When I tried to do the **Quickstart** from `README.md` the `configureExample.sh` failed because it couldn't find `picongpu/thirdParty/mallocMC`.

To fix this and make ParaTAXIS work even though PMacc and mallocMC steadily evolve a `git-submodule` should be added. For now in `CMakeLists.txt` the path `$HOME/picongpu` is set as `$PICONGPU_ROOT`. That environment variable should point towards the relative path of the `git-submodule` picongpu.

**Comment by n01r** [comment]
ParaTAXIS also seems to assume that its path is `~/parataxis`.

**Comment by ax3l** [comment]
Looks like no submodules involved. Try running `ccmake .` in your build dir after configuring.

I found in `CMakeLists.txt` that you can e.g. set the PIConGPU root to via `cmake -DPICONGPU_ROOT=somePath ...`

**Comment by n01r** [comment]
Yeah, sure, I contacted @Flamefire and he said that he forgot to add picongpu as a `git-submodule`. 
He worked with the most recent `dev` until he left.

**Comment by n01r** [comment]
**Update** The default `cmakeFlagsNr` is set to `-1` in `configureExample.sh`. But the compilation process requires the flag `-DPARAM_PRECISION` to be set. That would happen if the default were `flags[0]`. But as we've seen it is `-1` and nothing is set.

Thanks @psychocoderHPC for finding this.

**Comment by Flamefire** [comment]
Actually it was never planned to add submodules as this should at some point be included in PIConGPU. So I used the often used pattern by using a variable <PROGRAMM>_ROOT in the CMakeLists
Even more: This is only a helper. If there is a configured installation of MallocMC then it would be found w/o the picongpu path (https://github.com/ComputationalRadiationPhysics/parataxis/blob/8d24391df2171a73c6783ced3d32363df35e4fcf/CMakeLists.txt#L246)
However there was no way to find PMacc through modules so there it is required unless it can be found otherwise (https://github.com/ComputationalRadiationPhysics/parataxis/blob/8d24391df2171a73c6783ced3d32363df35e4fcf/CMakeLists.txt#L262)

So yes you can either install picongpu to the default location (which I set to be in the home) or point `PICONGPU_ROOT` to it.

About the flags: I think there was a change so that those defines have to be set instead of defaulting to something. So the default `cmakeFlagsNr` should be `0` I guess.

**Comment by ax3l** [comment]
Cool, can you add such information to `INSTALL.md` or somewhere prominent in the docs?
